README.md
Author does not hold any liability whatsoever for hair torn while reading, running or editing consequent script.

--GITHUB REPO: 
https://github.com/plasidu/phoenix4iraf  (Some sample output and input files are included)

--INTENDED USE: 
Script is intented for processing variable object(s) observations and it is meant to be extension of IRAF. Script takes the observed magnitudes and plot them agains time, which is given as heliocentric julian date in output graphs.

--DEPENDENCIES: 
Script requires installed vi(m), gnuplot, bash(-like environment with cat, wget and grep), python3 with numpy, jdcal and extra library from python2.7 (not requred for script itself) astrolibpy (https://code.google.com/p/astrolibpy/source/browse/astrolib/?r=b90f64233933fbe9392cc0e3c3ccaa6edf86696e) forked amateurishly into python3 version availible in github repo, and get_date.py in script's directory. In order to get objects coordinates fromSIMBAD database the computer has to be connected to internet. If this dependency is not met, user will be asked to enter the coordinates.

--INSTALLATION:
Install numpy and jdcal with pip or easy_install into python3 (sudo python3 -m easy_install jdcal).
Download phoenix4iraf.tar.gz from github.
Extract phoenix4iraf.tar.gz with tar -xzf phoenix4iraf.gz.tar. This creates working directory phoenix4iraf.
Cd into phoenix4iraf and make install and starting script executable with chmod 744 install start.
Then use sudo ./install .
That copies astrolibpy into python3 libraries directory at /usr/lib/python3/dist-packages/. WARNING: if your puthon3 is instaled elsewhere, copy astrolibpy into its libraries directory manualy and edit third line of ../astrolibpy/astrolib/helcorr.py so that it matches its current directory. Install then creates link into script's working directory (ln -s /usr/lib/python3/dist-package/astrolibpy/astrolib/helcorr.py . ). 

--INPUT: 
Input file should be in format "HH:MM:SS  MAG  ID", which is default for IRAF's "pdump files*mag* "otime, mag, id"", time is UT. One comparison object (C) and at least one check object (K) are required in data.

--STARTING THE SCRIPT: 
If all dependencies are met, run ./start . Optionaly, input file may be copied or linked to script's working directory. In the script itself, user input may be disabeled and variables may be set manualy. After running the script, files output.dat, plot_output.gp and various amount of .png graphs should be generated, as well as tmp directory, which is used for storing simbad output and is deleted in the process. Output.dat also include heliocentic correction in minutes.


--KNOWN BUGS: 
Data including midnight of from 29 Feb to 1 Mar will generate date 30 Feb instead.



2014 November 5, Yunnan observatory, plasidu at email dot cz


changelog
20141105:
-coordinates are downloaded from SIMBAD
-star_total determined from highest id
-increased verbosity


astrolibpy/astrolib/precess.py
from numpy import sin, cos, deg2rad, transpose, dot, arcsin, arctan2, zeros,\
                  ndarray, array, rad2deg, pi
from premat import premat

def precess(ra0, dec0, equinox1, equinox2, fk4=False, radian=False):
#   '''NAME:
#         PRECESS
#    PURPOSE:
#         Precess coordinates from EQUINOX1 to EQUINOX2.
#    EXPLANATION:
#         For interactive display, one can use the procedure ASTRO which calls
#         PRECESS or use the /PRINT keyword.   The default (RA,DEC) system is
#         FK5 based on epoch J2000.0 but FK4 based on B1950.0 is available via
#         the /FK4 keyword.
#   
#         Use BPRECESS and JPRECESS to convert between FK4 and FK5 systems
#    CALLING SEQUENCE:
#         PRECESS, ra, dec, [ equinox1, equinox2, /PRINT, /FK4, /RADIAN ]
#   
#    INPUT - OUTPUT:
#         RA - Input right ascension (scalar or vector) in DEGREES, unless the
#                 /RADIAN keyword is set
#         DEC - Input declination in DEGREES (scalar or vector), unless the
#                 /RADIAN keyword is set
#   
#         The input RA and DEC are modified by PRECESS to give the
#         values after precession.
#   
#    OPTIONAL INPUTS:
#         EQUINOX1 - Original equinox of coordinates, numeric scalar.  If
#                  omitted, then PRECESS will query for EQUINOX1 and EQUINOX2.
#         EQUINOX2 - Equinox of precessed coordinates.
#   
#    OPTIONAL INPUT KEYWORDS:
#         /PRINT - If this keyword is set and non-zero, then the precessed
#                  coordinates are displayed at the terminal.    Cannot be used
#                  with the /RADIAN keyword
#         /FK4   - If this keyword is set and non-zero, the FK4 (B1950.0) system
#                  will be used otherwise FK5 (J2000.0) will be used instead.
#         /RADIAN - If this keyword is set and non-zero, then the input and
#                  output RA and DEC vectors are in radians rather than degrees
#   
#    RESTRICTIONS:
#          Accuracy of precession decreases for declination values near 90
#          degrees.  PRECESS should not be used more than 2.5 centuries from
#          2000 on the FK5 system (1950.0 on the FK4 system).
#   
#    EXAMPLES:
#          (1) The Pole Star has J2000.0 coordinates (2h, 31m, 46.3s,
#                  89d 15' 50.6"); compute its coordinates at J1985.0
#   
#          IDL> precess, ten(2,31,46.3)*15, ten(89,15,50.6), 2000, 1985, /PRINT
#   
#                  ====> 2h 16m 22.73s, 89d 11' 47.3"
#   
#          (2) Precess the B1950 coordinates of Eps Ind (RA = 21h 59m,33.053s,
#          DEC = (-56d, 59', 33.053") to equinox B1975.
#   
#          IDL> ra = ten(21, 59, 33.053)*15
#          IDL> dec = ten(-56, 59, 33.053)
#          IDL> precess, ra, dec ,1950, 1975, /fk4
#   
#    PROCEDURE:
#          Algorithm from Computational Spherical Astronomy by Taff (1983),
#          p. 24. (FK4). FK5 constants from "Astronomical Almanac Explanatory
#          Supplement 1992, page 104 Table 3.211.1.
#   
#    PROCEDURE CALLED:
##          Function PREMAT - computes precession matrix
#   
#    REVISION HISTORY
#          Written, Wayne Landsman, STI Corporation  August 1986
#          Correct negative output RA values   February 1989
#          Added /PRINT keyword      W. Landsman   November, 1991
#          Provided FK5 (J2000.0)  I. Freedman   January 1994
#          Precession Matrix computation now in PREMAT   W. Landsman June 1994
#          Added /RADIAN keyword                         W. Landsman June 1997
#          Converted to IDL V5.0   W. Landsman   September 1997
#          Correct negative output RA values when /RADIAN used    March 1999
#          Work for arrays, not just vectors  W. Landsman    September 2003
#          Convert to PythonSergey KoposovJuly 2010 '''
   scal = True
   if isinstance(ra0, ndarray):
      ra = ra0.copy()  
      dec = dec0.copy()
      scal = False
   else:
      ra=array([ra0])
      dec=array([dec0])
   npts = ra.size 
   
   if not radian:   
      ra_rad = deg2rad(ra)     #Convert to double precision if not already
      dec_rad = deg2rad(dec)
   else:   
      ra_rad = ra
      dec_rad = dec
   
   a = cos(dec_rad)
   
   x = zeros((npts, 3))
   x[:,0] = a * cos(ra_rad)
   x[:,1] = a * sin(ra_rad)
   x[:,2] = sin(dec_rad)
   
   # Use PREMAT function to get precession matrix from Equinox1 to Equinox2
   
   r = premat(equinox1, equinox2, fk4=fk4)
   
   x2 = transpose(dot(transpose(r), transpose(x)))      #rotate to get output direction cosines
   
   ra_rad = zeros(npts) + arctan2(x2[:,1], x2[:,0])
   dec_rad = zeros(npts) + arcsin(x2[:,2])
   
   if not radian:   
      ra = rad2deg(ra_rad)
      ra = ra + (ra < 0.) * 360.e0            #RA between 0 and 360 degrees
      dec = rad2deg(dec_rad)
   else:   
      ra = ra_rad
      dec = dec_rad
      ra = ra + (ra < 0.) * 2.0e0 * pi
   
#   if do#print:   
      #print 'Equinox (%.2f): %f,%f' % (equinox2, ra, dec)
#   if scal:
#      ra, dec = ra[0], dec[0]
   return ra, dec



phoenix4iraf/start.sh
#!/usr/bin/env bash

# NOTE: input file should be in format "HH:MM:SS  MAG  ID", which is default for IRAF's "pdump files*mag* "otime, mag, id"", time is UT. Script is intented for processing variable object(s) observations, one comparison object (C) and at least one check object (K). 

# Uncomment 'read' to require user input
#read -p "Objects's right ascention (eq2000): " ra
#read -p "Objects's declination (eq2000): " dec
#read -p "UT date of the begining of observation (YYYY MM DD): " YYYY MM DD
#read -p "Total number of selected objects in data: " star_total
#read -p "Input file name: " input

# Uncomment and set variables for iterative task and/or debugging
ra=11.5
dec=17.3
YYYY=2014
MM=12
DD=31
star_total=3
input='phot_R.dat'

# Format the file for bash arrays
vim $input -c '%s/  / /g' -c '%s/:/ /g' -c 'x'

# Leap years are untreated
longmonths="01 03 05 07 08"
nextday=False
while read i
	do array=($i)
		if [ "${array[0]}" -eq 00 ] && [ "$nextday" == "False" ] && [ "$MM" -eq 12 ] && [ "$DD" -eq 31 ]; then
			DD=01
			MM=01
			YYYY=$(($YYYY + 1))
			nextday=True
		elif [ "${array[0]}" -eq 00 ] && [ "$nextday" == "False" ] && [ "$MM" -eq 10 ] && [ "$DD" -eq 31 ]; then
			DD=01
			MM=$(($MM + 1))
			nextday=True
		elif [ "${array[0]}" -eq 00 ] && [ "$nextday" == "False" ] && [ "$MM" -eq 02 ] && [ "$DD" -eq 28 ]; then
			DD=01
			MM=0$(($MM + 1))
			nextday=True
		elif [ "${array[0]}" -eq 00 ] && [ "$nextday" == "False" ] && [[ "$longmonths" =~ "$MM" ]] && [ "$DD" -eq 31 ]; then
			DD=01
			MM=0$(($MM + 1))
			nextday=True
		elif [ "${array[0]}" -eq 00 ] && [ "$nextday" == "False" ] && ! [[ "$longmonths" =~ "$MM" ]] && [ "$DD" -eq 30 ]; then
			DD=01
			MM=$(($MM + 1))
			nextday=True
		elif [ "${array[0]}" -eq 00 ] && [ "$nextday" == "False" ]; then
			DD=$(($DD + 1))
			nextday=True
		fi
		#python3 helcorr.py $ra $dec `python3 get_jd.py $YYYY $MM $DD ${array[0]} ${array[1]} ${array[2]}`
		hjd=`python3 get_jd.py $YYYY $MM $DD ${array[0]} ${array[1]} ${array[2]}`
		if [ "${array[4]}" -eq 1 ]; then
			printf "%s " `python3 helcorr.py $ra $dec $hjd`
			printf "${array[3]} "
		elif [ "${array[4]}" -eq "${star_total}" ]; then
			printf "${array[3]}\n"
		else
			printf "${array[3]} "
		fi
	done < $input > output.dat

# Plot output stream into separate png images
printf "set term png\nset xtics 0.1\nset output 'V-C.png'\nplot 'output.dat' u 1:(\$3-\$4) title 'V-C'\n" > plot_output.gp
for i in $(seq 1 $(($star_total-2)))
do printf "set output \'V-K"$i".png\'\nplot 'output.dat' u 1:(\$3-\$$(($i+4))) title 'V-K$i'\n" >> plot_output.gp
done

gnuplot plot_output.gp

